XNAControls/XNAScrollPanel.cs

```diff
@@ -0,0 +1,626 @@
+    
+    #region Properties
+    
+    #region Configurable properties
+    
+    /// <summary>
+    /// Whether to allow scrolling with arrow keys.
+    /// </summary>
+    public bool AllowKeyboardInput { get; set; } = true;
+
+    private (bool X, bool Y) _allowScroll = (true, true);
+    
+    /// <summary>
+    /// Whether to allow scrolling on X or Y axes.
+    /// </summary>
+    public (bool X, bool Y) AllowScroll
+    {
+        get => _allowScroll;
+        set
+        {
+            if (_allowScroll == value)
+                return;
+            
+            _allowScroll = value;
+            RecalculateScrollbars();
+        }
+    }
+
+    /// <summary>
+    /// How fast this control is scrolled?
+    /// </summary>
+    public int ScrollStep
+    {
+        get => HorizontalScrollBar.ScrollStep;
+        set => HorizontalScrollBar.ScrollStep 
+            = VerticalScrollBar.ScrollStep 
+                = value;
+    }
+    
+    private Point _overscrollMargin;
+
+    /// <summary>
+    /// How much can the content be "overscrolled", used for the purposes of inner panel size calculation.
+    /// </summary>
+    public Point OverscrollMargin
+    {
+        get => _overscrollMargin;
+        set
+        {
+            value = new(Math.Max(0, value.X), Math.Max(0, value.Y));
+            
+            if (_overscrollMargin == value)
+                return;
+            
+            ContentSize += value - _overscrollMargin;
+            _overscrollMargin = value;
+        }
+    }
+    
+    private bool _drawBorders;
+
+    public override bool DrawBorders
+    {
+        get => _drawBorders;
+        set
+        {
+            if (_drawBorders == value)
+                return;
+            
+            _drawBorders = value;
+            RecalculateScrollbars();
+        }
+    }
+
+    #endregion
+    
+    /// <summary>
+    /// Size of the inner panel that contains all the content.
+    /// </summary>
+    public Point ContentSize
+    {
+        get => ContentPanel.ClientRectangle.Size;
+        private set
+        {
+            ContentPanel.ClientRectangle = ContentPanel.ClientRectangle with { Size = value };
+            
+            HorizontalScrollBar.Length = value.X;
+            VerticalScrollBar.Length = value.Y;
+            
+            RecalculateScrollbars();
+        }
+    }
+    
+    /// <summary>
+    /// The physical offset of the <see cref="ContentPanel"/>.
+    /// </summary>
+    protected Point CurrentContentPanelPosition
+    {
+        get => ContentPanel.ClientRectangle.Location;
+        set
+        {
+            value = new Point
+            {
+                X = AllowScroll.X
+                    ? Math.Clamp(value.X, Math.Min(0, -(ContentSize.X - ViewSize.X)), 0)
+                    : ContentPanel.X,
+                Y = AllowScroll.Y
+                    ? Math.Clamp(value.Y, Math.Min(0, -(ContentSize.Y - ViewSize.Y)), 0)
+                    : ContentPanel.Y,
```

Comment:
Please, solve compilation errors for .net 4.8

<img width="605" height="254" alt="image" src="https://github.com/user-attachments/assets/95e910fe-27fa-4bd5-b17b-a58d8417d987" />